src/integration/NotificationOnEditHandler.spec.ts

```diff
@@ -0,0 +1,241 @@
+import express from "express"
+import request from "supertest"
+
+import { NotificationOnEditHandler } from "@middleware/notificationOnEditHandler"
+
+import UserSessionData from "@classes/UserSessionData"
+
+import {
+  Notification,
+  Repo,
+  Reviewer,
+  ReviewMeta,
+  ReviewRequest,
+  ReviewRequestView,
+  Site,
+  SiteMember,
+  User,
+  Whitelist,
+} from "@database/models"
+import { generateRouterForUserWithSite } from "@fixtures/app"
+import {
+  mockEmail,
+  mockIsomerUserId,
+  mockSiteName,
+} from "@fixtures/sessionData"
+import { GitHubService } from "@services/db/GitHubService"
+import * as ReviewApi from "@services/db/review"
+import { ConfigYmlService } from "@services/fileServices/YmlFileServices/ConfigYmlService"
+import { getUsersService, notificationsService } from "@services/identity"
+import CollaboratorsService from "@services/identity/CollaboratorsService"
+import IsomerAdminsService from "@services/identity/IsomerAdminsService"
+import SitesService from "@services/identity/SitesService"
+import ReviewRequestService from "@services/review/ReviewRequestService"
+import { sequelize } from "@tests/database"
+
+const mockSiteId = "1"
+const mockSiteMemberId = "1"
+
+const mockGithubService = {
+  getPullRequest: jest.fn(),
+  getComments: jest.fn(),
+}
+const usersService = getUsersService(sequelize)
+const reviewRequestService = new ReviewRequestService(
+  (mockGithubService as unknown) as typeof ReviewApi,
+  User,
+  ReviewRequest,
+  Reviewer,
+  ReviewMeta,
+  ReviewRequestView
+)
+const sitesService = new SitesService({
+  siteRepository: Site,
+  gitHubService: (mockGithubService as unknown) as GitHubService,
+  configYmlService: (jest.fn() as unknown) as ConfigYmlService,
+  usersService,
+  isomerAdminsService: (jest.fn() as unknown) as IsomerAdminsService,
+  reviewRequestService,
+})
+const collaboratorsService = new CollaboratorsService({
+  siteRepository: Site,
+  siteMemberRepository: SiteMember,
+  sitesService,
+  usersService,
+  whitelist: Whitelist,
+})
+
+const notificationsHandler = new NotificationOnEditHandler({
+  reviewRequestService,
+  collaboratorsService,
+  sitesService,
+  notificationsService,
+})
+
+// Set up express with defaults and use the router under test
+const subrouter = express()
+const subSubrouter = express()
+subSubrouter.get("/:siteName/test", async (req, res, next) =>
+  // Dummy subrouter
+  next()
+)
+subrouter.use(subSubrouter)
+
+// We handle the test slightly diferently - jest interprets the end of the test as when the response is sent,
+// but we normally create a notification after this response, due to the position of the middleware
+// the solution to get tests working is to send a response only after the notification middleware
+subrouter.use(async (req, res, next) => {
+  await notificationsHandler.createNotification(req as any, res as any, next)
```

Comment:
nit: await has no effect

Reply:
Hmm it seems to be required in the test, probably because jest seems to exit once the status code is received - I've left a comment explaining this in d6c1b3ea